pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -38,30 +38,201 @@ type Gatherer struct {
 	SecretLister             corelisters.SecretLister
 }
 
-// DataForCertificate returns the secret as well as the "current"
-// certificate request associated with the given certificate. It also
-// returns the given certificate as-is.
-//
-// The "current" certificate request designates the certificate request
-// that led to the current revision of the certificate. The "current"
-// certificate request is by definition in a ready state, and can be seen
-// as the source of information of the current certificate.
-//
-// This "current" certificate request is not to be confused with the "next"
-// certificate request that you might get by listing the CRs for the
-// certificate's revision+1; these "next" CRs might not be ready yet.
-//
-// We need the "current" certificate request because this CR contains the
-// "source of truth" of the current certificate, and getting the "current"
-// CR allows us to check whether the current certificate still matches the
+// DataForCertificate returns the secret as well as the "current" and "next"
+// certificate request associated with the given certificate. It also returns
+// the given certificate as-is.
+//
+// To understand the roles of the "current" and "next" certificate requests, let
+// us look at three different scenarii: A, B and C.
```

Comment:
@irbekrm @wallrj We might need diagrams to understand all this. I tried to rephrase all my comments 100 times but one can only do so much with only prose...

So I decided to cut the prose and instead have diagrams for the three main scenarii. 

I hope you find them useful to understand the finicky details about the next and current certificate request revisions 😅

Reply:
Can I vote for `scenarios`?

Reply:
I might have time too wordsmith the text on slack now-ish if you want to try.

Reply:
scenrii → scenarios fixed in ae4b5a63501342443677511ba3e75193af63081d 👍  Sorry for the delay